### ISSUE TITLE
ILC scanning failure for catch (Exception<T>) when using debug ILCompiler build

### ISSUE THREAD

**Issue by AntonLapounov** [dotnet/runtimelab#991]
Compiling DynamicGenerics test in release mode using a debug ILCompiler build produces the following error:
```
ILC : AOT analysis warning IL1005: GenericVirtualMethods.MyGenericTypeWithExceptionCatchingSupport`1<__Canon>.MyGenericTypeWithExceptionCatchingSupport`1<__Canon>(Boolean): Method will always throw because: Invalid IL or CLR metadata
ILC : AOT analysis warning IL1005: GenericVirtualMethods.MyGenericTypeWithExceptionCatchingSupport`1.<EnumerateStrings>d__1<__Canon>.MoveNext(): Method will always throw because: Invalid IL or CLR metadata
ILC : AOT analysis warning IL1005: GenericVirtualMethods.MyGenericTypeWithExceptionCatchingSupport`1.<AsyncMethod>d__4<__Canon>.MoveNext(): Method will always throw because: Invalid IL or CLR metadata
  *** Methods compiled but not scanned:
  [S.P.CoreLib]System.InvalidProgramException..ctor(string)
  [S.P.CoreLib]Internal.Runtime.TypeLoaderExceptionHelper.CreateInvalidProgramException(ExceptionStringID)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.ThrowHelpers.ThrowInvalidProgramException(ExceptionStringID)
  Unhandled exception. System.Exception: Scanning failure
```
Below is a simplified repro:
```C#
using System;

class Exception<T> : Exception { }

static class Repro
{
    static void Foo<T>()
    {
        try
        {
            Console.WriteLine("Hello");
        }
        catch (Exception<T>) { }
    }

    static void Main() { }
}
```
Build it with:
```cmd
dotnet publish -r win-x64 -c Release
```
@MichalStrehovsky

**Comment by MichalStrehovsky** [comment]
This regressed in https://github.com/dotnet/runtimelab/pull/536/commits/314ebe3123302c4f9728dd5dd2dd81607a9c8920.

What would probably fix this is to restore what was deleted in the pull request, while replacing the throwing part with `_dependencies.Add(_factory.MethodEntrypoint(_factory.TypeSystemContext.GetHelperEntrypoint("ThrowHelpers", "ThrowInvalidProgramException"), "Unsupported EH");`

**Comment by MichalStrehovsky** [comment]
Fixed in #1037
